chore: add sc e2e test

iamhopaul123 opened the pull request:
<!-- Provide summary of changes -->

<!-- Issue number, if available. E.g. "Fixes #31", "Addresses #42, 77" -->

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

iamhopaul123 committed:
chore: add sc e2e test

Lou1415926 commented on e2e/multi-svc-app/back-end/main.go:
```diff
@@ -23,17 +23,17 @@ func SimpleGet(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
-// ServiceDiscoveryGet just returns true no matter what
-func ServiceDiscoveryGet(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
-	log.Println("Get on ServiceDiscovery endpoint Succeeded")
+// ServiceConnectGet just returns true no matter what
+func ServiceConnectGet(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	log.Println("Get on service connect endpoint Succeeded")
 	w.WriteHeader(http.StatusOK)
-	w.Write([]byte("back-end-service-discovery"))
```
Instead of changing sd to sc, what do you think of keeping both in the e2e?
1. Deploy all services using SD, test that it works
2. Update all services to use SC, test that it works

iamhopaul123 committed:
Keep sd

iamhopaul123 committed:
Merge branch 'service-connect' into sc-e2e

iamhopaul123 committed:
Nit

iamhopaul123 committed:
Add back comment

KollaAdithya committed:
docs: add flowlog retention and autoscaling for release v1.24 (#4205)

Release v1.24 docs


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

iamhopaul123 committed:
Merge branch 'mainline' into sc-e2e

iamhopaul123 committed:
Fix paste error

huanjani commented on e2e/internal/client/outputs.go:
```diff
@@ -86,8 +87,8 @@ type SvcShowRoutes struct {
 
-// SvcShowServiceDiscoveries contains serialized service discovery info for an service.
-type SvcShowServiceDiscoveries struct {
+// SvcShowServiceEndpoints contains serialized endpoint info for an service.
```
tiny nit (not your typo):
```suggestion
// SvcShowServiceEndpoints contains serialized endpoint info for a service.
```

huanjani commented on e2e/multi-svc-app/back-end/main.go:
```diff
@@ -23,17 +23,17 @@ func SimpleGet(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
-// ServiceDiscoveryGet just returns true no matter what
-func ServiceDiscoveryGet(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
-	log.Println("Get on ServiceDiscovery endpoint Succeeded")
+// Get just returns true no matter what
```
another not-your-fault nit
```suggestion
// Get just returns true no matter what.
```

iamhopaul123 committed:
Addr nit

huanjani reviewed (approved):
lgtm
